modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js

```diff
@@ -374,7 +375,9 @@ const openSelectionModal = ({
 			if (searchContainer) {
 				iframeWindowObj.Liferay.componentReady(searchContainer.id).then(
 					(searchContainer) => {
-						const allSelectedElements = searchContainer.select.getAllSelectedElements();
+						const allSelectedElements = getSelectedItemsOnly
+							? searchContainer.select.getAllSelectedElements()
+							: searchContainer.select._getAllElements(false);
```

Comment:
`searchContainer.select._getAllElements(false)` returns not selected elements, right?

Which would mean the `allSelectedElements` variable they're being stored in has the wrong name, as it can store non-selected elements.

Reply:
`searchContainer.select._getAllElements(false)` doesn't return non-selected elements, **_it returns all elements, regardless of being selected or not_**.  See [the method](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js#L178-L185), [the method the previous method calls](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js#L198-L206), and [the specific logic the boolean argument is used in](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js#L203).

You can see when it's set to `true`, we retrieve all elements with `STR_CHECKED` (this is the "checked" attribute), and when set to false, we retrieve all elements, **period**.

Reply:
> ... it returns all elements, regardless of being selected or not.

That logic doesn't match the name `allSelectedElements`, right?

Reply:
`getAllSelectedElements()` = `getAllElements(true)`, see [this method](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/search_container_select.js#L260-L264).

`getAllElements(false)` actually lives up to the method name, because it is retrieving all the elements, regardless of checkbox status or not.

The confusing part is setting the argument to `true` returns only the checked boxes, but setting it to `false` doesn't return the non-checked boxes as seems intuitive.  It instead returns all checkboxes, regardless of status.  The easiest way to look at it, is if the argument was named `boolean onlySelected`, where `true` returns only selected checkboxes, and `false` returns all checkboxes.